spec/datetimeformat.html

```diff
@@ -168,8 +168,12 @@ <h1>ToDateTimeOptions ( _options_, _required_, _defaults_ )</h1>
       </p>
 
       <emu-alg>
-        1. If _options_ is *undefined*, let _options_ be *null*; otherwise let _options_ be ? ToObject(_options_).
-        1. Let _options_ be ObjectCreate(_options_).
+        1. If _options_ is *undefined*, then
+          1. Let _options_ be ObjectCreate(*null*).
+        1. Else,
+          1. If Type(_options_) is not Object,
+            1. Throw a *TypeError* exception.
+          1. Let _options_ be ObjectCreate(_options_).
```

Comment:
i'm kind of confused why ObjectCreate is/was used here - does the spec mutate the options object?

Reply:
I was trying to preserve the previous behaviour, so I don't know the answer to this; maybe because the subsequent Gets would otherwise be observable?

Reply:
ToDateTimeOptions does indeed perform some writes, and the consequences are unnecessarily weird.
```js
const wrappers = new Set();
const options = new Proxy({}, {
  get(target, prop, receiver){
    console.warn("get:", prop, (
      receiver === options ?
        "@options" :
      Object.getPrototypeOf(receiver) === options ?
        "@wrapper" + [...wrappers.add(receiver)].indexOf(receiver) :
      receiver
    ));
    let val = target[prop];
    Reflect.defineProperty(target, prop, {
      enumerable: true,
      get(){ return val },
      set(v){
        console.warn("set:", prop, v, (
          this === options ?
            "@options" :
          Object.getPrototypeOf(this) === options ?
            "@wrapper" + [...wrappers.add(this)].indexOf(this) :
          this
        ));
        val = v;
        return true;
      }
    });
    return val;
  }
});

new Intl.DateTimeFormat(undefined, options);

console.log([...wrappers]);
```

If we're making normative changes, I'd like to remove observable internal objects.

Reply:
In that case, shouldn’t the object be cloned before it’s written to, instead of making an inheriting object? (something that’s observable via proxies, and a very weird thing imo to have be observable)

Reply:
Long-overdue issue: #706